check/TypeCheck.hs

```diff
@@ -0,0 +1,50 @@
+module Main (main) where
+
+import Control.Monad (unless)
+
+import System.Directory (getCurrentDirectory)
+import System.Exit
+import System.IO
+import System.IO.Temp (withTempFile)
+import System.Process
+
+import Test.Hspec (Spec, describe, hspec, it)
+import Test.HUnit (assertFailure)
+
+import TypeDump (typeDump)
+
+main :: IO ()
+main = do
+    pwd <- getCurrentDirectory
+    hspec $ mapM_ (typeCheck pwd) modules
+
+modules :: [String]
+modules = [ "Prelude.Compat"
+          , "Control.Monad.Compat"
+          , "Data.Foldable.Compat"
+          , "Data.List.Compat"
+          ]
+
+typeCheck :: FilePath -> String -> Spec
+typeCheck pwd module_ = describe module_
+                      . it "should have the expected type signatures"
+                      . withTempFile pwd "Temp.types" $ \fp h -> do
```

Comment:
^ Personally I don't align in that way.  Firstly I only align by multiples of 2 (that would be for 4 for you I guess).  Secondly, you have to reallign everything if you e.g. add an other argument to `typeCheck` later on (or rename one, etc.).  My main issue with realligning is that you get bigger diffs.  This is harder to review and may lead to more conflicts.

For That reason, I would write the above code e.g. like so:

```haskell
typeCheck :: FilePath -> String -> Spec
typeCheck pwd module_ =
    describe module_
  . it "should have the expected type signatures"
  . withTempFile pwd "Temp.types" $ \fp h -> do
```